cmd/crc/cmd/delete.go

```diff
@@ -33,10 +33,6 @@ var deleteCmd = &cobra.Command{
 }
 
 func deleteMachine(client machine.Client, clearCache bool, cacheDir string, interactive, force bool) (bool, error) {
-	if !interactive && !force {
-		return false, errors.New("non-interactive deletion requires --force")
-	}
-
 	if clearCache {
 		yes := input.PromptUserForYesOrNo("Do you want to delete the OpenShift cluster cache", force)
```

Comment:
Before this `PromptUserForYesOrNo` line, the `!interactive && !force` check is required. It's probably the `checkIfMachineMissing(client)` check that you want to move earlier in this function.

Reply:
@cfergeau A user can clear the cache which is `~/.crc/cache` even the VM is not available so that's the reason I didn't move the machine check before.

Reply:
But `clearCache` is calling `input.PromptUserForYesOrNo`. Isn't it required to have the `!interactive && !force` check **before** any call to `input.PromptUserForYesOrNo`?

Reply:
@cfergeau yes, looks like it is like chicken-egg problem, I will take your suggestion to move the machine check in the beginning so at least it is not broken, just a trade-off.

Reply:
Or you could do the `!interactive && !force` check multiple times maybe?

Reply:
@cfergeau I will try to do something like so if user provide `--clear-cache` flag we can explicit check the interactive and force flags otherwise we will do it after machine exist check.

```golang
	if clearCache {
	        if !interactive && !force {
		        return false, errors.New("non-interactive deletion requires --force")
	        }
		yes := input.PromptUserForYesOrNo("Do you want to delete the OpenShift cluster cache", force)
		if yes {
			_ = os.RemoveAll(cacheDir)
		}
	}

	if err := checkIfMachineMissing(client); err != nil {
		return false, err
	}

        if !interactive && !force {
		return false, errors.New("non-interactive deletion requires --force")
	}
       yes := input.PromptUserForYesOrNo("Do you want to delete the OpenShift cluster", force)
	if yes {
		return true, client.Delete()
	}
	return false, nil

```